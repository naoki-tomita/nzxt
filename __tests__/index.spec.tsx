import { generate } from "../index";

describe("generate", () => {
  it("should generate html completely", async () => {
    const result = await generate("http://example.com/top");
    expect(result).toMatchSnapshot();
  });
});
