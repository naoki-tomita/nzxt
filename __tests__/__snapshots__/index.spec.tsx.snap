// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate should generate html completely 1`] = `
"<!DOCTYPE html><div ><span >document template</span><div id="nzxt-app"><div ><div >Top: 1</div><pre ># nzxt

A server side rendering framework for zheleznaya..

See [documentation](https://www.nzxt.tech/)
</pre><img src="/images/logo.png"></img></div></div><script >
    var parameter = {"star":1,"file":"# nzxt\\n\\nA server side rendering framework for zheleznaya..\\n\\nSee [documentation](https://www.nzxt.tech/)\\n"};
    function require(moduleName) {
      var errFn = function() { throw Error("This module is not callable on browser."); }
      var obj = new Proxy(errFn, { get(_, key) { return obj; } });
      return obj;
    };
    "use strict";(()=>{var G=Object.create;var w=Object.defineProperty;var J=Object.getOwnPropertyDescriptor;var Q=Object.getOwnPropertyNames;var X=Object.getPrototypeOf,Y=Object.prototype.hasOwnProperty;var Z=(e=>typeof require<"u"?require:typeof Proxy<"u"?new Proxy(e,{get:(t,r)=>(typeof require<"u"?require:t)[r]}):e)(function(e){if(typeof require<"u")return require.apply(this,arguments);throw Error('Dynamic require of "'+e+'" is not supported')});var l=(e,t)=>()=>(t||e((t={exports:{}}).exports,t),t.exports);var k=(e,t,r,n)=>{if(t&&typeof t=="object"||typeof t=="function")for(let i of Q(t))!Y.call(e,i)&&i!==r&&w(e,i,{get:()=>t[i],enumerable:!(n=J(t,i))||n.enumerable});return e};var j=(e,t,r)=>(r=e!=null?G(X(e)):{},k(t||!e||!e.__esModule?w(r,"default",{value:e,enumerable:!0}):r,e));var M=l(h=>{"use strict";Object.defineProperty(h,"__esModule",{value:!0});h.wrap=void 0;function T(e){if(e==null||typeof e!="object")return e;let t={__cb__:[],__emit__(){this.__cb__.forEach(n=>n())},__on__(n){this.__cb__.push(n)},__original__:e};Object.keys(e).forEach(n=>{let i=T(e[n]);e[n]=i,i?.__on__?.(()=>t.__emit__())});function r(n,i,o){let a=T(o);return e[i]=a,a?.__on__?.(()=>t.__emit__()),t.__emit__(),!0}return Array.isArray(e)?new Proxy(t,{get(n,i){return["__cb__","__emit__","__on__","__original__"].includes(i.toString())?n[i]:["push","pop","shift","unshift"].includes(i.toString())?(...o)=>{let a=e[i](...o);return t.__emit__(),a}:["map","forEach","reduce","filter","find","concat"].includes(i.toString())?(...o)=>e[i](...o):e[i]},set:r}):new Proxy(t,{get(n,i){return["__cb__","__emit__","__on__","__original__"].includes(i.toString())?n[i]:e[i]},set:r})}h.wrap=T});var P=l(p=>{"use strict";Object.defineProperty(p,"__esModule",{value:!0});p.range=p.isEquals=void 0;function x(e,t){if(typeof e!=typeof t)return!1;switch(typeof e){case"object":return e===null||t===null?e===t:Array.isArray(e)&&Array.isArray(t)?v(e,t):N(e,t);default:return e===t}}p.isEquals=x;function v(e,t){return e.length!==t.length?!1:$(e.length).every(r=>x(e[r],t[r]))}function N(e,t){let[r,n]=[Object.keys(e).sort(),Object.keys(t).sort()];return v(r,n)?r.every(i=>x(e[i],t[i])):!1}function $(e){return Array(e).fill(null).map((t,r)=>r)}p.range=$});var K=l(b=>{"use strict";Object.defineProperty(b,"__esModule",{value:!0});b.toKebabCaseFromSnakeCase=void 0;var ee="ABCDEFGHIJKLMNOPQRSTUVWXYZ";function te(e){return e.split("").map(t=>ee.includes(t)?\`-\${t.toLowerCase()}\`:t).join("")}b.toKebabCaseFromSnakeCase=te});var F=l(g=>{"use strict";Object.defineProperty(g,"__esModule",{value:!0});g.debounce=void 0;function re(e,t){let r;return(...n)=>{clearTimeout(r),r=setTimeout(()=>e(...n),t)}}g.debounce=re});var R=l(E=>{"use strict";Object.defineProperty(E,"__esModule",{value:!0});E.createEffect=void 0;var ne=P();function ie(){let e=["_____________________"];return function(r,n){(0,ne.isEquals)(e,n)||(r(),e=n)}}E.createEffect=ie});var q=l(c=>{"use strict";Object.defineProperty(c,"__esModule",{value:!0});c.createEffect=c.renderToText=c.render=c.useStore=c.getStore=c.createStore=c.h=void 0;var ue=M(),ce=P(),se=K(),oe=F();function ae(e,t,...r){return typeof e=="string"?{name:e,attributes:t,children:r,type:"html"}:()=>e(t,r)}c.h=ae;var y={};function H(e){return y=(0,ue.wrap)(e)}c.createStore=H;function fe(){return y}c.getStore=fe;function _e(e){return y.__on__?y:H(e)}c.useStore=_e;function z(e){return typeof e=="function"?C(e()):typeof e=="object"?Array.isArray(e)?{name:"ArrayNode",type:"array",attributes:{},children:e.map(t=>z(t))}:C(e):{name:e?.toString()??"",attributes:{},children:[],type:"text"}}function C(e){return typeof e=="function"&&(e=e()),{...e,children:(e.children||[]).map(t=>z(t))}}var _;function le(e,t){t&&(_=t),L(e),y.__on__?.(()=>pe(e))}c.render=le;var pe=(0,oe.debounce)(L,0);function me(e){let t=C(e);return A(t)}c.renderToText=me;function A(e){if(Array.isArray(e))return e.map(A).join("");switch(e.type){case"text":return e.name;case"html":return ye(e);case"array":return e.children.map(A).join("")}}function de(e){return typeof e=="string"?e:typeof e=="function"?"":Object.keys(e).map(t=>\`\${(0,se.toKebabCaseFromSnakeCase)(t)}: \${e[t]};\`).join("")}function ye(e){let t=null;if(typeof e.attributes?.ref=="function"){let r={};Object.defineProperty(r,"innerHTML",{set(n){t=n}}),e.attributes.ref(r)}return\`<\${e.name} \${Object.keys(e.attributes||{}).map(r=>\`\${r}="\${de(e.attributes[r])}"\`).join(" ")}>\${t??e.children.map(A).join("")}</\${e.name}>\`}var V=!0,I;function L(e){let t=C(e),r=Ce(t);I=r,V&&(_||(_=document.createElement("div"),document.body.appendChild(_)),_.innerHTML="",_.appendChild(r.element),V=!1)}function he(e){return{name:e.name,attributes:{},children:[],type:"text",element:document.createTextNode(e.name)}}function be(e,t,r){let n=[],i=[];return t?.type==="array"?t?.element?.forEach(o=>{r?.removeChild(o)}):t?.element&&r?.removeChild(t?.element),t&&(t.children=[]),e.children.forEach(o=>{let a=O(o,void 0,r);n.push(a.element),i.push(a)}),{name:"ArrayNode",attributes:{},type:"array",children:i,element:n}}var ge={input:["value","checked","disabled","readonly","required"]};function Ee(e,t,r){let n;if(t?.element!=null){if(n=t.element,!(0,ce.isEquals)(e.name,t.name)){let u=document.createElement(e.name);t.type==="array"?(t?.element?.forEach(s=>r?.removeChild(s)),r?.append(u)):r?.replaceChild(u,t.element),t.children=[],n=u}}else n=document.createElement(e.name);let{name:i,attributes:o}=e;Object.entries(o||{}).forEach(([u,s])=>{ge[i]?.includes(u)?n[u]=s:u==="style"?Object.entries(s).forEach(([d,f])=>n.style[d]=f):u.startsWith("on")?n[u]=s:typeof s=="boolean"?s?n.setAttribute(u,""):n.removeAttribute(u):u==="value"&&typeof n.value=="string"?n.value=s:u==="ref"&&typeof s=="function"?s(n):n.setAttribute(u,s)});let a=[];for(let u=0;u<e.children.length;u++){let s=e.children[u],d=t?.children[u],f=O(s,d,n);f.type==="array"?n.append(...f.children.map(B=>B.element)):d?.element||n.append(f.element),d?.type==="text"&&n.replaceChild(f.element,n.childNodes.item(u)),a.push(f)}return{...e,type:"html",element:n,children:a}}function O(e,t,r){switch(e.type){case"text":return he(e);case"array":return be(e,t,r);default:return Ee(e,t,r)}}function Ce(e){return O(e,I,_)}var Ae=R();Object.defineProperty(c,"createEffect",{enumerable:!0,get:function(){return Ae.createEffect}})});var S=j(q());var m=j(q());var U=Z("fs/promises"),D=({star:e,file:t})=>(0,m.h)("div",null,(0,m.h)("div",null,"Top: ",e),(0,m.h)("pre",null,t),(0,m.h)("img",{src:"/images/logo.png"}));D.getInitialPrpos=async()=>{let e=await fetch("https://api.github.com/repos/naoki-tomita/nzxt").then(r=>r.json()).then(r=>r.stargazers_count),t=await(0,U.readFile)("./README.md").then(r=>r.toString("utf-8"));return{star:e,file:t}};var W=D;(function(e){(0,S.render)((0,S.h)(W,{...e}),document.getElementById("nzxt-app"))})(parameter);})();

  </script></div>"
`;
